wgpu-hal/src/metal/adapter.rs

```diff
@@ -925,7 +925,7 @@ impl super::PrivateCapabilities {
         features.set(
             F::TEXTURE_BINDING_ARRAY
                 | F::SAMPLED_TEXTURE_AND_STORAGE_BUFFER_ARRAY_NON_UNIFORM_INDEXING
-                | F::UNIFORM_BUFFER_AND_STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING
+                | F::STORAGE_TEXTURE_ARRAY_NON_UNIFORM_INDEXING
```

Comment:
similar to vulkan, shouldn't the `F::UNIFORM_BUFFER_INDEXING` be set somewhere here as well? In wgpu-types you specified `with MSL 2.0+ on macOS 10.13+` as the condition